File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def getIntersectionNode(self, headA, headB):
        curr1 = headA
        curr2 = headB
        l1 = []
        l2 = []
        while curr1:   
            l1.append(curr1)
            curr1 = curr1.next
        while curr2:
            l2.append(curr2)
            curr2 = curr2.next
        for i in range(len(l1)):
            for j in range(len(l2)):
                if l1[i]==l2[j]:
                    return l1[i]
        
        return None

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode *getIntersectionNode(struct ListNode *headA, struct ListNode *headB) {
    struct ListNode* la = headA;
    struct ListNode* lb = headB;
    int lenA = 0;
    int lenB = 0;
    while (la) {
        lenA++;
        la = la->next;
    }
    while (lb) {
        lenB++;
        lb = lb->next;
    }

    int diff = 0;
    if (lenA > lenB) {
        diff = lenA-lenB;
        struct ListNode* currA = headA;
        struct ListNode* currB = headB;
        while (diff > 0) {
            currA = currA->next;
            diff--;
        }

        while (currA && currB) {
            if (currA == currB) return currA;
            currA = currA->next;
            currB = currB->next;
        } 
    }

    else {
        diff = lenB-lenA;
        struct ListNode* currA = headA;
        struct ListNode* currB = headB;
        while (diff > 0) {
            currB = currB->next;
            diff--;
        }

        while (currA && currB) {
            if (currA == currB) return currA;
            currA = currA->next;
            currB = currB->next;
        } 
    }

    return NULL;
}